File: Step registration/ClientApp/src/app/models/country.ts
import { Province } from "./province";

export class Country {

  constructor(id: string, name: string, provinces: Province[]) {
    this.id = id;
    this.name = name;
    this.provinces = provinces;
  }

  id!: string;
  name!: string;
  provinces!: Province[];
}

File: DB/Models/Country.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StepRegistration.Models
{
    public class Country
    {
        [Key]
        public Guid Id { get; set; }

        public virtual string Name { get; set; }

        public Country()
        {
            Id = Guid.NewGuid();
            Name = string.Empty;
        }

        [ForeignKey("CountryId")]
        public virtual IList<Province> Provinces { get; set; }
    }
}
